File: patch2.py
import sys

with open("src/app/draft-room/page.tsx", "r") as f:
    orig_code = f.read()

start_sig = "  const processedPool = useMemo(() => {"
end_sig = "  }, [displayPool, searchTerm, showDrafted, draftResults, positionFilter, statSort, yahooStats, yahooPlayers, activeSport]);"

start_idx = orig_code.find(start_sig)
end_idx = orig_code.find(end_sig) + len(end_sig)

pool_hook = orig_code[start_idx:end_idx] + "\n"
code = orig_code[:start_idx] + orig_code[end_idx:]

# Find displayPool end
dp_end_sig = "  }, [searchTerm, myRoster, draftResults, showDrafted, positionFilter, aiNotes]);"
dp_end_idx = code.find(dp_end_sig) + len(dp_end_sig)

if dp_end_idx == len(dp_end_sig) - 1:
    print("Failed to find dp_end_sig")
    sys.exit(1)

code = code[:dp_end_idx] + "\n\n" + pool_hook + code[dp_end_idx:]

with open("src/app/draft-room/page.tsx", "w") as f:
    f.write(code)

print("Patched successfully via python.")

File: patch2.js
const fs = require('fs');
let code = fs.readFileSync('src/app/draft-room/page.tsx', 'utf8');

// Find the processedPool hook
const poolHookStart = "  const processedPool = useMemo(() => {";
const poolHookEnd = "  }, [displayPool, searchTerm, showDrafted, draftResults, positionFilter, statSort, yahooStats, yahooPlayers, activeSport]);";

const startIndex = code.indexOf(poolHookStart);
const endIndex = code.indexOf(poolHookEnd) + poolHookEnd.length;

if (startIndex !== -1 && endIndex !== -1) {
    const poolHook = code.slice(startIndex, endIndex) + "\\n";
    code = code.slice(0, startIndex) + code.slice(endIndex);

    // Find displayPool and insert it after
    const displayPoolEnd = "  }, [searchTerm, myRoster, draftResults, showDrafted, positionFilter, aiNotes]);\\n";
    const dpEndIndex = code.indexOf(displayPoolEnd) + displayPoolEnd.length;

    if (dpEndIndex !== -1 + displayPoolEnd.length) {
        code = code.slice(0, dpEndIndex) + "\\n" + poolHook + code.slice(dpEndIndex);
        fs.writeFileSync('src/app/draft-room/page.tsx', code);
        console.log("Moved processedPool after displayPool successfully.");
    } else {
        console.error("Could not find displayPool block end");
    }
} else {
    console.error("Could not find processedPool hook");
}
